environment.cpp

```diff
@@ -127,15 +137,20 @@ double dummy_function(std::vector<double> input)
 
 void switch_env_function(environment &e)
 {
-  if(are_same_env_functions(e.get_current_function(), e.get_env_function_A())){
+  if(e.get_name_current_function()=='A'){
       e.change_env_function(e.get_env_function_B());
+      e.switch_name_current_function();
     }
   else
```

Comment:
maybe use an `else if` to be more precise and use the else to stop everything and throw an exception. It might happen that function name is something absurd, (nor A nor B), and we want to catch that as well.